File: server/src/services/ai/geminiClient.js
import { GoogleGenAI } from '@google/genai';
import config from '../../config.js';
import logger from '../../utils/logger.js';

let _ai = null;

function getAi() {
  if (!_ai) {
    if (!config.GEMINI_API_KEY || config.GEMINI_API_KEY === 'your_api_key_here') {
      throw new Error('GEMINI_API_KEY not set in .env — add your real API key');
    }
    _ai = new GoogleGenAI({ apiKey: config.GEMINI_API_KEY });
    logger.info('Gemini AI client initialized');
  }
  return _ai;
}

// Proxy that lazily initializes the client on first use
export const ai = new Proxy(
  {},
  {
    get(_, prop) {
      return getAi()[prop];
    },
  }
);

export const models = config.models;

File: server/app/services/ai/gemini_client.py
from google import genai

from app.config import config
from app.logger import logger

_client: genai.Client | None = None


def get_client() -> genai.Client:
    global _client
    if _client is None:
        if not config.GEMINI_API_KEY or config.GEMINI_API_KEY == "your_api_key_here":
            raise RuntimeError("GEMINI_API_KEY not set in .env — add your real API key")
        _client = genai.Client(api_key=config.GEMINI_API_KEY)
        logger.info("Gemini AI client initialized")
    return _client


models = config.models
